doc/srv6/srv6_sid_l3adj.md

```diff
@@ -0,0 +1,124 @@
+
+## Scope  
+
+Extending the SRv6Orch to support the programming of the L3Adj associated with SRv6 uA, End.X, uDX4, uDX6, End.DX4, and End.DX6 behaviors.
+
+## Definitions/Abbreviations 
+
+| **Term**                 | **Definition**                                     |
+|--------------------------|----------------------------------------------------|
+| End.X                    | L3 Cross-Connect                                   |
+| End.DX4                  | Decapsulation and IPv4 Cross-Connect               |
+| End.DX6                  | Decapsulation and IPv6 Cross-Connect               |
+| L3Adj                    | Layer 3 Adjacency                                  |
+| SID                      | Segment Routing Identifier                         |
+| SRv6                     | Segment Routing over IPv6                          |
+| uA                       | End.X behavior with NEXT-CSID, PSP and USD flavors |
+| uDX4                     | End.DX4 behavior with NEXT-CSID flavor             |
+| uDX6                     | End.DX4 behavior with NEXT-CSID flavor             |
+
+## Overview 
+
+The support of SRv6 has been defined in this HLD: [Segment Routing over IPv6 (SRv6) HLD](https://github.com/sonic-net/SONiC/blob/master/doc/srv6/srv6_hld.md).
+
+The HLD includes several SRv6 behaviors defined in [RFC 8986](https://datatracker.ietf.org/doc/html/rfc8986).
+The Appl DB was extended to support these behaviors. 
+
+The Appl DB includes the L3Adj attributes which is used with these behaviors: uA, End.X, uDX4, uDX6, End.DX4, and End.DX6.
+
+The current implementation of SRv6Orch does not process the L3Adj.
+
+In this HLD, we extend the SRv6Orch to process the L3Adj and program it in the ASIC DB.
+
+## High-Level Design 
+
+The following diagram shows the SRv6Orch workflow to process an SRv6 SID associated with a L3Adj in SONiC:
+<br> <br>
+
+![SRv6 SID L3Adj Sequence Diagram](images/srv6_sid_l3adj_sequence_diagram.png "SRv6 SID_L3Adj Sequence Diagram").
+
+- SRv6Orch is an APPL_DB subscriber.
+- SRv6Orch receives a `SRV6_MY_SID_TABLE` update notification about the SID. The SID is associated with a L3 adjacency carried in the `adj` parameter.
+- SRv6Orch gets the nexthop ID of the adjacency from NeighOrch.
+- SRv6Orch sets the nexthop ID attribute of the SID.
+- SRv6Orch invokes the sairedis `sai_srv6_api->create_my_sid_entry()` API to create the SRv6 SID entry in the ASIC DB. 
+
+The next subsections describe the SRv6Orch changes required to support the HLD described above.
+
+### SRv6Orch Changes
+
+We extend SRv6Orch to support the programming of the L3Adj associated with uA, End.X, uDX4, uDX6, End.DX4, and End.DX6 behaviors.
+
+When SRv6Orch receives a SID with the `adj` parameter set, it calls the function `neighOrch->hasNextHop()` to make sure a nexthop associated with the adjacency exists.
+- If the nexthop does not exist, SRv6Orch returns an error.
```

Comment:
How about ECMP? Does Fpmsyncd change(https://github.com/sonic-net/sonic-swss/pull/2515/files) handle ECMP as well? Please update accordingly.

Reply:
@cscarpitta Could you update the doc with ECMP for MYSID entries.

Reply:
@ kperumalbfn Hi Kumaresh, the scope of the PR will be limited to a single L3Adj. We are doing of the required changes to support ECMP and we are targeting 202405 for the ECMP support. Thanks!